crates/store/re_grpc_client/src/connection_client.rs

```diff
@@ -0,0 +1,183 @@
+                id: Some(entry_id.into()),
+            })
+            .await?;
+
+        Ok(())
+    }
+
+    pub async fn create_dataset_entry(
+        &mut self,
+        name: String,
+    ) -> Result<DatasetEntry, StreamError> {
+        let response: CreateDatasetEntryResponse = self
+            .inner()
+            .create_dataset_entry(CreateDatasetEntryRequest { name: Some(name) })
+            .await?
+            .into_inner()
+            .try_into()?;
+
+        Ok(response.dataset)
+    }
+
+    pub async fn read_dataset_entry(
+        &mut self,
+        entry_id: EntryId,
+    ) -> Result<DatasetEntry, StreamError> {
+        let response: ReadDatasetEntryResponse = self
+            .inner()
+            .read_dataset_entry(ReadDatasetEntryRequest {
+                id: Some(entry_id.into()),
+            })
+            .await?
+            .into_inner()
+            .try_into()?;
+
+        Ok(response.dataset_entry)
+    }
+
+    pub async fn update_dataset_entry(
+        &mut self,
+        entry_id: EntryId,
+        dataset_details: DatasetDetails,
+    ) -> Result<DatasetEntry, StreamError> {
+        let response: UpdateDatasetEntryResponse = self
+            .inner()
+            .update_dataset_entry(tonic::Request::new(
+                UpdateDatasetEntryRequest {
+                    id: entry_id,
+                    dataset_details,
+                }
+                .into(),
+            ))
+            .await?
+            .into_inner()
+            .try_into()?;
+
+        Ok(response.dataset_entry)
+    }
+
+    /// Get a list of partition IDs for the given dataset entry ID.
+    pub async fn get_dataset_partition_ids(
+        &mut self,
+        entry_id: EntryId,
+    ) -> Result<Vec<PartitionId>, StreamError> {
+        const COLUMN_NAME: &str = ScanPartitionTableResponse::PARTITION_ID;
+
+        let mut stream = self
+            .inner()
+            .scan_partition_table(tonic::Request::new(
+                ScanPartitionTableRequest {
+                    dataset_id: entry_id,
+                    scan_parameters: Some(ScanParameters {
+                        columns: vec![COLUMN_NAME.to_owned()],
+                        on_missing_columns: IfMissingBehavior::Error,
+                        ..Default::default()
+                    }),
+                }
+                .into(),
+            ))
+            .await?
+            .into_inner();
+
+        let mut partition_ids = Vec::new();
+
+        while let Some(resp) = stream.next().await {
+            let record_batch = resp?.data()?.decode()?;
+
+            let partition_id_col = record_batch
+                .column_by_name(COLUMN_NAME)
+                .ok_or_else(|| StreamError::MissingDataframeColumn(COLUMN_NAME.to_owned()))?;
+
+            let partition_id_array =
+                partition_id_col.try_downcast_array_ref::<arrow::array::StringArray>()?;
+
+            partition_ids.extend(
```

Comment:
collecting this to a vec means that we're blocking until we're through with gathering all the partitions. Not a problem now, but sounds like something we'd have to fix up later. Any chance we can return an async iterator here?